### ISSUE TITLE
Implementation on PyTorch for Windows 10

### ISSUE THREAD

**Issue by MaximilianBoemer** [Eromera/erfnet_pytorch#7]
Hi Eromera,
first thanks for your inspiring work! I wanted to recreate your project with a conda package of PyTorch for Windows 10 x64, Anaconda3 (Python 3.6), Cuda 8.0 and pytorch-0.3.0. When I tried to start the training I receive an IndexError because a list index is out of range. The same error also appears, when I try to evaluate the trained model on the validation set.

![train error_li](https://user-images.githubusercontent.com/28864847/33896953-79f9bb3a-df64-11e7-9ec6-b7805677b2cc.jpg)

![evaluate error_li](https://user-images.githubusercontent.com/28864847/33896931-6ee48068-df64-11e7-816f-1f111fe1b927.jpg)

Do you think this kind of errors are quick to fix or are they appearing because I try to deploy it on Windows?

All the best,
Max

**Comment by Eromera** [comment]
Hi! Thanks for your message. That's interesting, I haven't tried anything in Windows and you seem to be the first so you are kinda "alone in the dark" ;) 

Despite of that, the error seems relatively easy, at least it seems related to the dataset loading. I think the operation throws an "index out of range" error because the list of images is empty. I would debug if the path is being correctly loaded. First check if the path you used as argument is correct (and all the data is correctly arranged as loaded by the code), then I would try to debug (e.g. print) what is the data path that the code sees before loading that list. Some things might have been hardcoded to work only in Ubuntu unintendedly (like using "/" as separation between folders), if you localize the errors I'll be happy to fix them and update the code to be more general to Windows.  Anyway I'll try to have a look at all the code as soon as I have time. Cheers!

**Comment by MaximilianBoemer** [comment]
Thanks for the super quick reply! I will start tomorrow with this and keep you updated!

**Comment by rob93chan** [comment]
I think I also had this error (but not on Windows 10) and changed line 17 in [train/dataset.py](https://github.com/Eromera/erfnet_pytorch/blob/master/train/dataset.py) to `return filename.endswith("_labelIds.png")`. Maybe it works for you to overcome this error.

**Comment by Eromera** [comment]
Hi @MaximilianBoemer , just a quick addition that might be related to your problem. I added the following info in the README regarding the dataset since there has been some confusion in the past, I dunno if you were using labelIds or trainIds:

> Cityscapes Dataset: Download the "leftImg8bit" for the RGB images and the "gtFine" for the labels. Please note that for training you should use the "_labelTrainIds" and not the "_labelIds", you can download the cityscapes scripts and use the conversor to generate trainIds them from labelIds.

**Comment by Eromera** [comment]
Hi @MaximilianBoemer , any news with the Windows implementation or did you desist and use Ubuntu? 😝 I might try to make the code compatible with Windows in the near future. Thanks!

**Comment by MaximilianBoemer** [comment]
Hi @Eromera I am sorry for not giving news. Actually I stopped working on it and implemented an other architecture now. Thanks for your help again!

**Comment by Eromera** [comment]
Thanks for the update, then I'll close the issue for now, feel free to reopen anytime!
